Android/res/layout/fragment_mode_auto.xml

```diff
@@ -1,10 +1,92 @@
 <?xml version="1.0" encoding="utf-8"?>
-<TextView
+<LinearLayout
     xmlns:android="http://schemas.android.com/apk/res/android"
-    android:id="@+id/ModeDetail"
-    style="@style/ModeDetailText"
-    android:background="@drawable/mode_desc_rectangle"
     android:layout_width="match_parent"
-    android:layout_height="wrap_content"
-    android:text="@string/mode_auto"/>
+    android:layout_height="match_parent"
+    android:gravity="center_horizontal"
+    android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/ModeDetail"
+        style="@style/ModeDetailText"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/mode_desc_rectangle"
+        android:text="@string/mode_auto" />
+
+    <ProgressBar
+        android:id="@+id/mission_progress"
+        style="?android:attr/progressBarStyleHorizontal"
+        android:layout_width="match_parent"
+        android:layout_height="20dp"
+        android:layout_marginBottom="10dp"
+        android:animationResolution="@android:integer/config_shortAnimTime"
+        android:indeterminate="false"
+        android:max="100"
+        android:progress="0"
+        android:progressBackgroundTint="@android:color/holo_blue_dark"
+        android:progressTint="@android:color/holo_blue_dark" />
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="48dp"
+        android:layout_margin="5dp"
+        android:background="@drawable/button_blue"
+        android:gravity="center"
+        android:orientation="horizontal">
+
+        <com.andexert.library.RippleView
```

Comment:
@TShapinsky the `LinearLayout` children should have the `layout_weight` attribute so they can stretch to fill out all the space within their parent is necessary.